policy/daemon_test.go

```diff
@@ -938,6 +944,242 @@ func Test_policyd_CheckPolicy(t *testing.T) {
+							return a
+						}(),
+					})
+					return g
+				}(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				domain: "dummyDom",
+				roles:  []string{"dummyRole", "dummyRole1"},
+				// It doesn't match.
+				action:   "dummyAct1234",
+				resource: "dummyRes",
+			},
+			want: errors.New("no match: Access denied due to no match to any of the assertions defined in domain policy file"),
+		},
+		{
+			name: "check can use wildcard on resource",
+			fields: fields{
+				rolePolicies: func() gache.Gache {
+					g := gache.New()
+					g.Set("dummyDom:role.dummyRole", []*Assertion{
+						func() *Assertion {
+							a, _ := NewAssertion("dummyAct", "dummyDom:*Res?", "allow")
+							return a
+						}(),
+					})
+					return g
+				}(),
+			},
+			args: args{
+				ctx:      context.Background(),
+				domain:   "dummyDom",
+				roles:    []string{"dummyRole", "dummyRole1"},
+				action:   "dummyAct",
+				resource: "dummyRes1",
+			},
+			want: nil,
+		},
+		{
+			name: "check can use wildcard on resource, deny",
+			fields: fields{
+				rolePolicies: func() gache.Gache {
+					g := gache.New()
+					g.Set("dummyDom:role.dummyRole", []*Assertion{
+						func() *Assertion {
+							a, _ := NewAssertion("dummyAct", "dummyDom:*Res?", "allow")
+							return a
+						}(),
+					})
+					return g
+				}(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				domain: "dummyDom",
+				roles:  []string{"dummyRole", "dummyRole1"},
+				// It doesn't match.
+				action:   "dummyAct",
+				resource: "dummyRes",
+			},
+			want: errors.New("no match: Access denied due to no match to any of the assertions defined in domain policy file"),
+		},
+		{
+			name: "check can use wildcard on action and resource",
+			fields: fields{
+				rolePolicies: func() gache.Gache {
+					g := gache.New()
+					g.Set("dummyDom:role.dummyRole", []*Assertion{
+						func() *Assertion {
+							a, _ := NewAssertion("*Act?", "dummyDom:*Res?", "allow")
+							return a
+						}(),
+					})
+					return g
+				}(),
+			},
+			args: args{
+				ctx:      context.Background(),
+				domain:   "dummyDom",
+				roles:    []string{"dummyRole", "dummyRole1"},
+				action:   "dummyAct1",
+				resource: "dummyRes1",
+			},
+			want: nil,
+		},
+		{
+			name: "check can use wildcard on action and resource, deny",
+			fields: fields{
+				rolePolicies: func() gache.Gache {
+					g := gache.New()
+					g.Set("dummyDom:role.dummyRole", []*Assertion{
+						func() *Assertion {
+							a, _ := NewAssertion("*Act?", "dummyDom:*Res?", "allow")
+							return a
+						}(),
+					})
+					return g
+				}(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				domain: "dummyDom",
+				roles:  []string{"dummyRole", "dummyRole1"},
+				action: "dummyAct1",
+				// It doesn't match.
+				resource: "dummyRes1234",
+			},
+			want: errors.New("no match: Access denied due to no match to any of the assertions defined in domain policy file"),
+		},
```

Comment:
Please add a test case about `"\*"`. Escaping wildcard is not supported.

Reply:
Fixed bfbaadd